src/services/decoders/ClpIrDecoder.ts

```diff
@@ -1,32 +1,61 @@
-import clpFfiJsModuleInit, {ClpIrStreamReader} from "clp-ffi-js";
-
+import clpFfiJsModuleInit, {ClpStreamReader} from "../../../deps/ClpFfiJs-worker";
 import {Nullable} from "../../typings/common";
 import {
+    ClpIrDecoderOptions,
     Decoder,
     DecodeResultType,
     FilteredLogEventMap,
     LogEventCount,
 } from "../../typings/decoders";
-import {LogLevelFilter} from "../../typings/logs";
+import {JsonObject} from "../../typings/js";
+import {
+    LOG_LEVEL,
```

Comment:
I found a way to avoid the usage of `ts-expect-error`:
```
            // `this.#formatter` has been null-checked at the method entry.
            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
            result[0] = this.#formatter!.formatLogEvent({
                fields: fields,
                level: level,
                timestamp: dayJsTimestamp,
            });
```
so we can still inline the argument object creation with the type checks in effect, which looks more concise in my opinion. Let me know if you prefer to have variable definition separated.